homeassistant/components/tuya/__init__.py

```diff
@@ -120,29 +120,36 @@ async def async_setup(hass, config):
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
     """Set up Tuya platform."""
 
-    tuya = TuyaApi()
+    tuya = hass.data.get(DOMAIN, {}).get(TUYA_DATA)
```

Comment:
Probably it is better to add a comment, as `#Try to get tuya object created in config flow`